programs/perpetuals/src/state/oracle.rs

```diff
@@ -49,6 +50,24 @@ pub struct CustomOracle {
 
 impl CustomOracle {
     pub const LEN: usize = 8 + std::mem::size_of::<CustomOracle>();
+
+    pub fn set(&mut self, price: u64, expo: i32, conf: u64, ema: u64, publish_time: i64) {
+        self.price = price;
+        self.expo = expo;
+        self.conf = conf;
+        self.ema = ema;
+        self.publish_time = publish_time;
+    }
+
+    pub fn serialize(&self) -> [u8; 36] {
```

Comment:
anchor uses borsch to serialize. by adding AnchorSerialize to the struct, you can serialize without writing this method

Reply:
ahh this was fun!  previously I had rolled the encoding format by just stacking the fields.

Updated contract-side and client-side to use `borsh`.  It also cleaned up the code on both ends 👍